File: SERVICIO WEB CON SERVLETS, Y HUFFMAN/PAGINA WEB SERVLETS/js/quicksort.js
function quicksort()
{
    this.ordena = function(pivote, sublista)
    {
        if(sublista.length < 2)
            return sublista;
        var listMenores = new Array();
        var puntPivote = sublista.splice(pivote, 1)[0];
        for(i=sublista.length-1; i>=0; i--)
            if(sublista[i].p < puntPivote.p)
                listMenores.push(sublista.splice(i, 1)[0]);
        return this.ordena(parseInt(sublista.length/2), sublista).concat(puntPivote).concat(this.ordena(parseInt(listMenores.length/2), listMenores));
    };
}

File: SERVICIO WEB CON SERVLETS, Y HUFFMAN/ARCHIVOS FUENTE JAVA/quicksort.java
import java.util.ArrayList;

public class quicksort {
    public ArrayList<nodo> ordena(int pivote, ArrayList<nodo> sublista)
    {
        if(sublista.size() < 2)
            return sublista;
        ArrayList<nodo> listMenores = new ArrayList<nodo>();
        nodo puntPivote = sublista.get(pivote);
        sublista.remove(pivote);
        for(int i=sublista.size()-1; i>=0; i--)
            if(sublista.get(i).peso < puntPivote.peso)
            {
                listMenores.add(sublista.get(i));
                sublista.remove(i);
            }
        ArrayList<nodo> retorno = ordena(sublista.size()/2, sublista);
        retorno.add(puntPivote);
        retorno.addAll(ordena(listMenores.size()/2, listMenores));
        return retorno;
    }
}
